### PR TITLE
add a basic integration test

### PR DESCRIPTION
**Pull request by RuaidhriMcDonnell**
As requested by @cjfarrelly. Without some integration tests, its impossible for him to create valid evo release pipelines.
This simple test will cover that requirement.

The environment.cs file is based off the logic checkout / logistics have been using in their projects.

### CODE REVIEW

File: src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace WebAPIService.Integration.Tests
+{
+    [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+    public static class Environment
+    {
+        static Environment()
+        {
+            var isBuildServer = System.IO.File.Exists(System.IO.Path.Combine(System.Environment.CurrentDirectory, "appsettings.INTEGRATION.json"));
+            Name = System.Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? string.Empty;
```
Comment (VaclavK) [inline src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs]: I think we have helper in the devopssdk for this

File: src/content/src/Tests/WebAPIService.Integration.Tests/appsettings.json [hunk 2]
```diff
@@ -1,2 +1,5 @@
 ﻿{
+  "Endpoints": {
+    "WebAPIServiceApi": "http://localhost:23217"
```
Comment (VaclavK) [inline src/content/src/Tests/WebAPIService.Integration.Tests/appsettings.json]: ??? should this be tied to port number passed to the template?

File: src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+using Microsoft.Extensions.Configuration;
+
+namespace WebAPIService.Integration.Tests
+{
+    [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+    public static class Environment
+    {
+        static Environment()
+        {
+            var isBuildServer = System.IO.File.Exists(System.IO.Path.Combine(System.Environment.CurrentDirectory, "appsettings.INTEGRATION.json"));
```
Comment (VaclavK) [inline src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs]: also would refer to devops sdk

File: src/content/src/Tests/WebAPIService.Integration.Tests/appsettings.json (hunk 2 above)
Comment (RuaidhriMcDonnell) [reply src/content/src/Tests/WebAPIService.Integration.Tests/appsettings.json]: it should, but the current template doesn't support it right now, and it needs to apply globally.

Comment (VaclavK) [reply src/content/src/Tests/WebAPIService.Integration.Tests/appsettings.json]: the template recognizes APIPortNumber and it is used in the manifest - Endpoint Protocol="http" Name="ServiceEndpoint" Type="Input" Port="WebAPIServicePort" />

File: src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs (hunk 1 above)
Comment (VaclavK) [reply src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs]: https://github.com/eShopWorld/devops/blob/915367c9970e614a80d30da8069bc81fb0c93788/src/Eshopworld.DevOps/EswDevOpsSdk.cs#L95

File: src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs (hunk 3 above)
Comment (VaclavK) [reply src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs]: would not the buildconfiguration with test mode set to true work for you?

File: src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs (hunk 1 above)
Comment (RuaidhriMcDonnell) [reply src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs]: ok. this requires the packages in the template to be updated, which i'd rather not do in this PR.
if people are happy with the initial changes, and me merge to master, a subsequent PR can be made to use this new functionality, which would include the package bumps

File: src/content/src/Tests/WebAPIService.Integration.Tests/appsettings.json (hunk 2 above)
Comment (RuaidhriMcDonnell) [reply src/content/src/Tests/WebAPIService.Integration.Tests/appsettings.json]: different port number. there are 2 port numbers. IISExpress (generally used by devs) hosted application and then the SF hosted application port

File: src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs (hunk 3 above)
Comment (RuaidhriMcDonnell) [reply src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs]: you tell me. this code was the code that was handed to the checkout team by devops at the time. i'm only reusing what the current (near to production) standard is

Comment (dguerin) [inline src/content/src/Tests/WebAPIService.Integration.Tests/Environment.cs]: yeah I'd probably use a cleaner version of this if there is one. I think one of the purposes of the template is illustrate best practice so using the devops sdk version if there is one or a cleaner way would be better.

**Comment by dguerin** [comment]
Closing per offline convos

**Pull request closed without merging**